src/main/kotlin/com/zepben/evolve/cim/iec61970/base/core/IdentifiedObject.kt

```diff
@@ -62,30 +62,75 @@ abstract class IdentifiedObject(mRID: String = "") {
      */
     fun getName(type: String, name: String): Name? = _names?.getByTypeAndName(type, name)
 
-    fun addName(name: Name): IdentifiedObject {
-        require(name.identifiedObject === this) { "Attempting to add a Name to ${typeNameAndMRID()} that does not reference this identified object" }
+    /**
+     * The individual name information of the identified object.
+     *
+     * @param name the name of the required [Name]
+     * @param type the required [NameType]
+     * @return The [Name] with the specified [name] if it exists, otherwise null
+     */
+    fun getName(type: NameType, name: String): Name? = _names?.getByTypeAndName(type.name, name)
+
+    /**
+     * All name information of the identified object given a [NameType].
+     *
+     * @param type the required [NameType]
+     * @return List of [Name] with the specified [type] if it exists, otherwise null
+     */
+    fun getNames(type: NameType): List<Name>? = _names?.filter { it.type == type }
```

Comment:
todo: I believe this function does not behave as described in the docstring (I have not tested it).

The docstring indicates it returns `null` if there is no `Name` of the specified `type`, however I am pretty sure it actually returns the following:

1. A list of `Name` instances if there are matches.
2. `null` if there are no names at all.
3. An empty list if there are names, but none that match the `NameType`

Can you please add a test to confirm and make the return values consistent if I am correct.

Reply:
Done